src/components/constructor-form/constructor-form.js

```diff
@@ -0,0 +1,347 @@
+
+  useEffect(() => {
+    dispatch(
+      getMaterials({
+        filter
+      })
+    );
+    if (editableConstructorElement) {
+      setConstructorImg(editableConstructorElement.image);
+    }
+  }, [dispatch, filter]);
+
+  const languageTabs =
+    languages.length > 0
+      ? languages.map((lang) => (
+        <Tab label={lang} data-cy={`${lang}-tab`} key={lang} />
+      ))
+      : null;
+
+  const constructorValidationSchema = Yup.object().shape({
+    enName: Yup.string()
+      .min(2, CONSTRUCTOR_VALIDATION_ERROR)
+      .required(CONSTRUCTOR_ERROR_MESSAGE),
+    uaName: Yup.string()
+      .min(2, CONSTRUCTOR_VALIDATION_ERROR)
+      .required(CONSTRUCTOR_ERROR_MESSAGE),
+    material: Yup.string()
+      .required(CONSTRUCTOR_ERROR_MESSAGE),
+    image: Yup.string().required(PHOTO_NOT_PROVIDED),
+    basePrice: Yup.string()
+      .matches(config.formRegExp.onlyPositiveDigits, PRICE_VALIDATION_ERROR)
+      .required(CONSTRUCTOR_ERROR_MESSAGE)
+  });
+
+  const {
+    values,
+    handleSubmit,
+    handleChange,
+    touched,
+    errors,
+    setFieldValue
+  } = useFormik({
+    validationSchema: constructorValidationSchema,
+    initialValues: {
+      image: editableConstructorElement.image || '',
+      uaName: editableConstructorElement.name[0].value || '',
+      enName: editableConstructorElement.name[1].value || '',
+      material: editableConstructorElement.material._id || '',
+      available: editableConstructorElement.available || false,
+      basePrice: +editableConstructorElement.basePrice[1].value / 100 || 0
+    },
+
+    onSubmit: () => {
+      const constructorElement = createConstructor(values);
+      history.goBack()
+      if(isEdit){
+        return dispatch(constructorElementMethod({
+          constructorElement,
+          id:editableConstructorElement._id
+        }))
+      }
+      return dispatch(constructorElementMethod({
+        constructorElement,
+        id:model._id
+      }))
+    }
+  });
+
+  const handleMaterial = (e) => {
+    setFieldValue('material', e.target.value)
+  }
+  const checkboxes = [
+    {
+      id: 'available',
+      dataCy: 'available',
+      value: values.available,
+      checked: values.available,
+      color: 'primary',
+      label: config.labels.model.show,
+      handler: () => setFieldValue('available', !values.available)
+    }
+  ];
+
+  const handleImageLoad = (e, callback) => {
+    if (e.target.files && e.target.files[0]) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        callback(event);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+    }
+  };
+
+  const handleLoadConstructorImage = (e) => {
+    handleImageLoad(e, (event) => {
+      setFieldValue('image', event.target.result);
+      setConstructorImg(event.target.result);
+    });
+  };
+
+  const textField = (inputValue, inputLabel) => (<TextField
+    data-cy={inputValue}
+    id={inputValue}
```

Comment:
id cannot be input value. in default case it will be empty

TextFiled should have name or id related to Formik fields

Reply:
Fixed